dataqs/helpers.py

```diff
@@ -381,3 +381,21 @@ def __init__(self, content):
 
     def raise_for_status(self):
         pass
+
+
+def add_keywords(keyword_list, extra_keywords):
+    """Adds extra_keywords list to the keyword_list"""
+
+    # check if datetime and category already exist in the
+    # keyword list
+
+    keywords_added = [k for k in keyword_list
+                      if k.startswith('category:') or k.startswith('datetime:')]
+
+    if keywords_added:
+        filtered_keywords = [k for k in keyword_list
+                             if not k.startswith('category:') or
+                             not k.startswith('datetime:')]
```

Comment:
Check your boolean algebra, `not (a or b) <=> not a and not b`.  Also do you really need different conditions here?  Can't you just always return `filtered_keywords + extra_keywords`?